src/Storages/System/StorageSystemTables.cpp

```diff
@@ -455,6 +459,38 @@ class TablesBlockSource : public ISource
                         res_columns[res_index++]->insertDefault();
                 }
 
+                auto table_merge_tree = std::dynamic_pointer_cast<MergeTreeData>(table);
+                if (columns_mask[src_index++])
+                {
+                    if (table_merge_tree)
+                        res_columns[res_index++]->insert(table_merge_tree->getAllDataPartsVector().size());
+                    else
+                        res_columns[res_index++]->insertDefault();
+                }
+
+                if (columns_mask[src_index++])
+                {
+                    if (table_merge_tree)
+                        res_columns[res_index++]->insert(table_merge_tree->getPartsCount());
+                    else
+                        res_columns[res_index++]->insertDefault();
+                }
+
+                if (columns_mask[src_index++])
+                {
+                    if (table_merge_tree)
+                    {
+                        size_t total_marks = 0;
+                        for (auto & part : table_merge_tree->getAllDataPartsVector())
+                        {
+                            total_marks += part->getMarksCount();
+                        }
```

Comment:
That for also could be made as member of MergeTreeData. In that case no allocation temp vector, no coping shared_ptrs.